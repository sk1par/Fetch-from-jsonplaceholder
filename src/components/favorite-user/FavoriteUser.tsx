import React from 'react'
import IUser from '../../models/user.interface'

interface User {
    user: IUser;
};

const FavoriteUser = ({user}: User) => {
    return (
        <div className="col-auto mb-3">
            <div className="card" style={{width: '18rem'}}>
                <div className="card-body">
                    <h5 className="card-title">Name:{user?.name}</h5>
                    <h6 className="card-subtitle mb-2 text-muted">Username: {user?.username}</h6>
                    <p className="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>
                    <a href={"tel:" + user?.phone} className="card-link">{user?.phone}</a><br />
                        <a href={"mailto:" + user?.email} className="card-link">{user?.email}</a>
                </div>
            </div>
        </div>
    )
}

export default FavoriteUser