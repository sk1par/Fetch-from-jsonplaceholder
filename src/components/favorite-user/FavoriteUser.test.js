import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import FavoriteUser from './FavoriteUser';

Enzyme.configure({ adapter: new Adapter() });

it('should render user details  div', () => {
    const wrapper = shallow(<FavoriteUser />);
    const userDetailDiv = wrapper.find('[data-auto-id="user-detail-div"]');
    expect(userDetailDiv).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});
