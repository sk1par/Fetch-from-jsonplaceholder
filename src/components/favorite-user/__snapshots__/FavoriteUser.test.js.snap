// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render user details  div 1`] = `ShallowWrapper {}`;
